test/e2e_new/kubetest/test.go

```diff
@@ -0,0 +1,168 @@
+	Expect(os.MkdirAll(outputDir, 0750)).To(Succeed(), "Invalid test suite argument. Can't create output directory %q", outputDir)
+	ginkgoVars := map[string]string{
+		"nodes":             strconv.Itoa(c.GinkgoNodes),
+		"slowSpecThreshold": strconv.Itoa(c.GinkgoSlowSpecThreshold),
+	}
+	e2eVars := map[string]string{
+		"kubeconfig":           proxy.GetKubeconfigPath(),
+		"provider":             "skeleton",
+		"report-dir":           c.reportDir(),
+		"e2e-output-dir":       outputDir,
+		"dump-logs-on-failure": "false",
+		"report-prefix":        "kubetest.",
+		"num-nodes":            strconv.FormatInt(numNodes, 10),
+		"viper-config":         c.ConfigFilePath,
+	}
+	ginkgoArgs := buildArgs(ginkgoVars, "-")
+	e2eArgs := buildArgs(e2eVars, "--")
+	e2eCmd := exec.Command("ginkgo", ginkgoArgs...)
+	e2eCmd.Args = append(e2eCmd.Args, c.e2ePath())
+	e2eCmd.Args = append(e2eCmd.Args, "--")
+	e2eCmd.Args = append(e2eCmd.Args, e2eArgs...)
+	e2eCmd.Dir = c.KubernetesRepoRoot
+	e2eCmd = completeCommand(e2eCmd, "Running e2e test", false)
+	e2eErr := e2eCmd.Run()
+	// Rename files for consumption by prow
+	Expect(GatherReports(c)).To(Succeed())
+	Expect(e2eErr).NotTo(HaveOccurred())
+}
+
+func (c *Configuration) reportDir() string {
+	return path.Join(c.ArtifactsDirectory, "kubetest")
+}
+
+func GatherReports(c *Configuration) error {
+	if err := os.MkdirAll(c.reportDir(), 0750); err != nil {
+		return err
+	}
+	return filepath.Walk(c.reportDir(), func(p string, info os.FileInfo, err error) error {
+		if info.IsDir() && p != c.reportDir() {
+			return filepath.SkipDir
+		}
+		if filepath.Ext(p) != ".xml" {
+			return nil
+		}
+		base := filepath.Base(p)
+		if strings.HasPrefix(base, "junit") {
+			newName := strings.ReplaceAll(base, "_", ".")
+			if err := os.Rename(p, path.Join(filepath.Dir(p), "..", newName)); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+
+}
+
+// buildArgs converts a string map to the format --key=value
+func buildArgs(kv map[string]string, flagMarker string) []string {
+	args := make([]string, len(kv))
+	i := 0
+	for k, v := range kv {
+		args[i] = flagMarker + k + "=" + v
+		i++
+	}
+	return args
+}
+
+// buildK8sPrereqs builds kubectl and kubetest
+func (c Configuration) buildK8sPrereqs() {
+	cmd := exec.Command("bazel", "build", "//cmd/kubectl", "//test/e2e:e2e.test", "//vendor/github.com/onsi/ginkgo/ginkgo")
+	cmd.Dir = c.KubernetesRepoRoot
+
+	cmd = completeCommand(cmd, "Ensuring e2e build prerequisites", false)
+	err := cmd.Run()
+	Expect(err).NotTo(HaveOccurred())
+}
+
+// copyK8sPrereqs copies bazel outputs to a location for consumption by gingko
+func (c Configuration) copyK8sPrereqs() {
+	binDir := path.Join(c.KubernetesRepoRoot, "_output", "bin")
+	Expect(os.MkdirAll(binDir, 0750)).To(Succeed(), "Can't create output directory")
+	err := copyFile(path.Join(c.KubernetesRepoRoot, "/bazel-bin/test/e2e/e2e.test"), path.Join(binDir, "e2e.test"))
+	Expect(err).NotTo(HaveOccurred())
+}
```

Comment:
It would be great if we could avoid building the dependencies..  @dims is the e2e.test binary included in the ci-artifacts, and if not, could we get them added?

Reply:
@detiber yes we could get them added, please open an issue in k/k and ping me.

Reply:
@detiber seems like we have the conformance image uploaded!! https://console.cloud.google.com/gcr/images/kubernetes-ci-images/GLOBAL/conformance@sha256:cd04049a2288ec515c6435f805982a2bce9e08a7133e2dd99fa515a74ebe4d3d/details?tab=info

that should have everything needed to run e2e tests

Reply:
example scripts are there in https://github.com/kubernetes/kubernetes/tree/master/cluster/images/conformance

Reply:
omg, that is amazing, that also likely means that we could leverage sonobuoy to run the tests/gather the results

Reply:
thanks @dims. switching over to sonobuoy as I write.